Automattic/wp-calypso
# Domains: Update instructions for mapping subdomains

#### Changes proposed in this Pull Request

Continuing the changes done in #51031 and #50668, this PR updates the **sub**domain mapping instructions that are presented to users. These instructions are shown in the thank you page, after the user buys the subdomain mapping, and in the domains management page, when a subdomain isn't mapped yet.

This PR presents CNAME or NS records mappings as recommended setups depending if the site is simple or AT, respectively. Alternative mapping setups (e.g. using A records) are shown as advanced setup options.

#### Testing instructions



Firstly the D59270-code patch should be applied.

Both simple and atomic sites should be tested.

1. In Calypso, go to Upgrades > Domains > Add a domain to this site
2. Select "Use a domain I own"
3. "Map your domain"
4. Insert a subdomain
5. Buy that subdomain mapping for your site
6. The "Thank you" page with further instructions to setup subdomain mapping will be shown.

The instructions should differ if the site is simple or atomic.

* For simple sites: The recommended setup should be to set a CNAME record pointing to the user's WPCOM domain, and there should be two advanced setup options:
  * The first should instruct the user to set NS records pointing to our name servers.
  * The second should show the default IP addresses (*.24 and *.25) to be set as A records;

<img width="582" alt="subdomain mapping simple 1_" src="https://user-images.githubusercontent.com/5324818/113192783-95dad280-9235-11eb-8acc-9a538528b276.png">

* For AT sites: The recommended setup should be to set NS records pointing to our name servers, and there should be two advanced setup options:
  * The first should instruct the user to set a CNAME record pointing to their WPCOM staging domain.
  * The second should show the AT site's IP addresses to be set as A records;

<img width="584" alt="subdomain mapping AT 1_" src="https://user-images.githubusercontent.com/5324818/113192798-9d01e080-9235-11eb-81e2-c099214a88e9.png">

CNAME mapping for AT sites should show a warning:

<img width="585" alt="subdomain mapping AT 2_" src="https://user-images.githubusercontent.com/5324818/113192811-a12dfe00-9235-11eb-8cbc-fc6bd09bd8c8.png">
